src/cli/rustup_mode.rs

```diff
@@ -1334,11 +1335,23 @@ fn target_remove(cfg: &Cfg, m: &ArgMatches) -> Result<utils::ExitCode> {
     let distributable = DistributableToolchain::try_from(&toolchain)?;
 
     for target in m.get_many::<String>("target").unwrap() {
-        let new_component = Component::new(
-            "rust-std".to_string(),
-            Some(TargetTriple::new(target)),
-            false,
-        );
+        let target = TargetTriple::new(target);
+        let default_target = cfg.get_default_host_triple()?;
+        if target == default_target {
+            warn!("after removing the default host target, proc-macros and build scripts might no longer build");
+        }
+        let has_at_most_one_target = {
+            let components = distributable.components()?;
+            // Every component target that is not `None` (wildcard).
+            let targets = components
+                .iter()
+                .filter_map(|c| c.installed.then(|| c.component.target.clone()).flatten());
+            targets.unique().at_most_one().is_ok()
+        };
```

Comment:
What do you think about this alternative?

```rust
let has_at_most_one_target = distributable
            .components()?
            .into_iter()
            .filter_map(|c| match (c.installed, c.component.target) {
                (true, Some(t)) => Some(t),
                _ => None,
            })
            .unique()
            .at_most_one()
            .is_ok();
```

This is a bit longer (in formatted line count), but it avoids the nested combinators and the need for cloning the target triples.